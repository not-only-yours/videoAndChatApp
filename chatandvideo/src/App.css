.app {
  display: grid;
  place-items: center;
  background-color: lightgrey;
  height: 100vh;
}

.app_body {
  display: flex;
  margin-top: -30px;
  background-color: #ededed;
  height: 90vh;
  width: 90vw;
  box-shadow: -1px 4px 20px -6px black;
}
