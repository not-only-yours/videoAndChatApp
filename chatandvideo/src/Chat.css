.chat {
  flex: 0.65;
  display: flex;
  flex-direction: column;
}

.chat_header {
  padding: 20px;
  display: flex;
  align-items: center;
  border-bottom: 1px solid lightcyan;
}

.chat_headerInfo {
  flex: 1;
  padding-left: 20px;
}

.chat_headerInfo > h3 {
  margin: 0 3px 0 0;
  font-weight: 500;
}

.chat_headerInfo > p {
  margin: 0;
  color: gray;
}

.chat_headerRight {
  display: flex;
  justify-content: space-between;
  min-width: 100px;
}

.chat_body {
  overflow: scroll;
  flex: 1;
  background-color: lightblue;
}

.chat_message {
  margin-left: 20px;
  margin-top: 5px;
  position: relative;
  font-size: 16px;
  padding: 10px;
  border-radius: 10px;
  background-color: white;
  width: fit-content;
  margin-bottom: 15px;
}

.chat_name {
  position: absolute;
  top: 0;
  font-weight: 800;
  font-size: xx-small;
}

.chat_timestamp {
  margin-left: 10px;
  font-size: xx-small;
}

.chat_reciever {
  margin-right: 20px;
  margin-left: auto;
  background-color: lightgreen;
}

.chat_footer {
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-top: 1px solid lightgray;
  height: 62px;
}

.chat_footer > form {
  flex: 1;
  display: flex;
}

.chat_footer > form > input {
  flex: 1;
  border-radius: 30px;
  padding: 10px;
  border: none;
}

.chat_footer > form > button {
  display: none;
}
