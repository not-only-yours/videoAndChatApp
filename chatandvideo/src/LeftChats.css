.leftpart_chat {
  display: flex;
  padding: 5px;
  cursor: pointer;
  border-bottom: 1px solid grey;
}

a {
  text-decoration: none !important;
  color: black;
}

.leftpart_chat:hover {
  background-color: lightgrey;
}

.leftpartChat_info > h2 {
  font-size: 16px;
  margin-left: 20px;
}

.leftpartChat_info > p {
  font-size: 14px;
  margin-left: 20px;
}
