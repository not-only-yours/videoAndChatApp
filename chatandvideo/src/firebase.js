import firebase from "firebase";
const firebaseConfig = {
  apiKey: "AIzaSyD3OrFbwg35jgpDfrCUnSauNwWjvB8D29A",
  authDomain: "chatpart-18f0f.firebaseapp.com",
  databaseURL: "https://chatpart-18f0f-default-rtdb.firebaseio.com",
  projectId: "chatpart-18f0f",
  storageBucket: "chatpart-18f0f.appspot.com",
  messagingSenderId: "15268948799",
  appId: "1:15268948799:web:97b729bdf7a62cf8b2818a",
  measurementId: "G-PYZBV7J98Z",
};

const firebaseApp = firebase.initializeApp(firebaseConfig);

const db = firebaseApp.firestore();
export const auth = firebase.auth();
export const provider = new firebase.auth.GoogleAuthProvider();

export default db;
