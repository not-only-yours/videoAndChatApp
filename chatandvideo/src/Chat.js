import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";
import "./Chat.css";
import { Avatar, IconButton } from "@material-ui/core";
import { VideoCall } from "@material-ui/icons";
import db from "./firebase";

function Chat() {
  const [input, setInput] = useState("");
  const [roomName, setRoomName] = useState("");
  const { roomId } = useParams();

  useEffect(() => {
    if (roomId) {
      db.collection("rooms")
        .doc(roomId)
        .onSnapshot((snapshot) => setRoomName(snapshot.data().name));
    }
  }, [roomId]);

  const sendMessage = (e) => {
    e.preventDefault();
    console.log(input);

    setInput("");
  };

  return (
    <div className="chat">
      <div className="chat_header">
        <Avatar />
        <div className="chat_headerInfo">
          <h3>{roomName}</h3>
          <p>Last seen at ...</p>
        </div>
        <div className="chat_headerRight">
          <IconButton>
            <VideoCall />
          </IconButton>
        </div>
      </div>
      <div className="chat_body">
        <p className={`chat_message ${true && "chat_reciever"}`}>
          <span className="chat_name">SSSSSSssssss</span>
          hey everybody!
          <span className="chat_timestamp">2:12am</span>
        </p>
      </div>
      <div className="chat_footer">
        <form>
          <input
            value={input}
            onChange={(e) => setInput(e.target.value)}
            type="text"
            placeholder="Type a message..."
          />
          <button onClick={sendMessage} type="submit">
            Send a message
          </button>
        </form>
      </div>
    </div>
  );
}

export default Chat;
