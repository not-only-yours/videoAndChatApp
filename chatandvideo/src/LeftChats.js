import React from "react";
import "./LeftChats.css";
import { Avatar } from "@material-ui/core";
import { Link } from "react-router-dom";
import { createRoom, idExists } from "./service";

function LeftChats({ id, name, addProp }) {
  const [messages, setMessages] = React.useState("");
  React.useEffect(() => {
    if (id) {
      idExists(id, setMessages);
    }
  }, [id]);
  const createChat = () => {
    const roomName = prompt("Введіть назву чатіку");
    if (roomName) {
      // alert(`RoomName: ${roomName}`);
      // створити в firebase базу і пушити сюди її
      createRoom(roomName);
    }
  };
  return !addProp ? (
    <Link to={`/rooms/${id}`}>
      <div className="leftpart_chat">
        <Avatar />
        <div className="leftpartChat_info">
          <h2>{name}</h2>
          <p>{messages[0]?.message}</p>
        </div>
      </div>
    </Link>
  ) : (
    <div onClick={createChat} className="leftpart_chat">
      <h2>Добавити новий чатік</h2>
    </div>
  );
}

export default LeftChats;
