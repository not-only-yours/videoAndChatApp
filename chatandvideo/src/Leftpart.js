import React, { useState, useEffect } from "react";
import "./Leftpart.css";
import { IconButton } from "@material-ui/core";
import AccessibleIcon from "@material-ui/icons/Accessible";
import SearchIcon from "@material-ui/icons/Search";
import InfoIcon from "@material-ui/icons/Info";
import LeftChats from "./LeftChats";
import { useBetween } from "use-between";

export const useShareableState = () => {
  const [rooms, setRooms] = useState([
    {
      id: 1,
      name: "Chat1",
    },
    {
      id: 2,
      name: "Chat2",
    },
  ]);
  const [count, setCount] = useState(3);
  return {
    rooms,
    setRooms,
    count,
    setCount,
  };
};

function Leftpart() {
  const { rooms } = useBetween(useShareableState);
  return (
    <div className="leftpart">
      <div className="leftpart_header">
        <AccessibleIcon />
        <div className="leftpart_headerRight">
          <IconButton>
            <SearchIcon />
          </IconButton>
          <IconButton>
            <InfoIcon />
          </IconButton>
        </div>
      </div>
      <div className="leftpart_search">
        <div className="leftpart_searchContainer">
          <SearchIcon />
          <input type="text" placeholder="Print here dude" />
        </div>
      </div>
      <div className="leftpart_chats">
        <LeftChats addProp />
        {rooms.map((room) => (
          <LeftChats key={room.id} id={room.id} name={room.name} />
        ))}
      </div>
    </div>
  );
}

export default Leftpart;
