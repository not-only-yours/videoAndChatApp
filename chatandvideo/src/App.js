import "./App.css";
import Leftpart from "./Leftpart";

function App() {
  return (
    <div className="app">
      <div className="app_body">
        <Leftpart />
        {/*Тут буде вікно самого чату */}
      </div>
    </div>
  );
}

export default App;
