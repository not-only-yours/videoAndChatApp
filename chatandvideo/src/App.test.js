import LeftChats from "./LeftChats";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow, configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Leftpart from "./Leftpart";
import Login from "./Login";
import App from "./App";
import Video from "./video";
import { createSerializer } from "enzyme-to-json";
import renderer from "react-test-renderer";
import React from "react";
import Chat from "./Chat";
import {
  send,
  jwtExists,
  roomNameExists,
  sendMessageFun,
  idExists,
  createRoom,
  refreshDB,
} from "./service";
import { StateContext, StateProvider } from "./StateProvider";
import reducer, { initialState } from "./reducer";
import "firebase/firestore";
import FirestoreMock from "./mockedFirebase";
import db from "./firebase";
import IconButton from "@material-ui/core/IconButton";
import sinon from "sinon";

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));
configure({ adapter: new Adapter() });
const firestoreMock = new FirestoreMock();

beforeEach(() => {
  const mockSetState = jest.fn();
  jest.mock("./StateProvider", () => ({
    useStateValue: () => ["initial", mockSetState],
  }));
  jest.mock("react", () => ({
    useState: (initial) => [
      [
        { room: initial, id: "asa", data: { name: "aaa" } },
        { room: initial, id: "asa", data: { name: "aaa" } },
      ],
      mockSetState,
    ],
    useContext: (initial) => [initial, mockSetState],
    useRef: jest.fn(),
    useEffect: jest.fn(),
    useReducer: (initial, setter) => [initial, setter],
    createContext: jest.fn(),
  }));
  jest.mock("./Chat", () => ({
    handleSubmit: jest.fn().mockResolvedValue("test"),
  }));
  jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: () => ({
      roomId: "group1",
    }),
  }));
  jest.mock("./service", () => ({
    jwtExists: jest.fn(),
    sendMessageFun: jest.fn(),
    send: jest.fn().mockResolvedValue("test"),
    idExists: jest.fn(),
    createRoom: jest.fn(),
    refreshDB: jest.fn(),
    signIn: jest.fn().mockResolvedValue("test"),
    twillioConnect: (token, videoRoomName, localVidRef, remoteVidRef) => "aaa",
  }));
});
describe("test modules with router", () => {
  it("test module LeftChats with snapshot", () => {
    const container = renderer
      .create(
        <Router>
          <LeftChats />
        </Router>
      )
      .toJSON();
    expect(container).toMatchSnapshot();
  });

  it("test module Video with snapshot", () => {
    const container = renderer.create(<Video />).toJSON();
    expect(container).toMatchSnapshot();
  });
});

describe("shallow tests of modules", () => {
  it("testing LeftChats with parameters key, id, name", () => {
    const room = {
      id: true,
      data: {
        name: "name",
      },
    };
    const container = shallow(
      <LeftChats id={room.id} name={room.data.name} addProp={false} />
    );
    expect(container).toMatchSnapshot();
  });

  it("testing LeftChats with addProp", () => {
    const container = shallow(<LeftChats addProp />);
    expect(container).toMatchSnapshot();
  });

  it("testing Chat", () => {
    const container = shallow(<Chat />);
    container.find("button").simulate("click", { preventDefault: () => {} });
    expect(container).toMatchSnapshot();
  });

  it("testing Leftpart", () => {
    const container = shallow(<Leftpart />);
    expect(container).toMatchSnapshot();
  });

  it("testing Login", () => {
    const container = shallow(<Login />);
    expect(container).toMatchSnapshot();
  });
  it("app without user", () => {
    const container = shallow(<App />);
    expect(container).toMatchSnapshot();
  });

  it("app with user, token", () => {
    jest.resetModules();
    jest.mock("./StateProvider", () => ({
      useStateValue: () => [{ user: "adas", token: "sas" }, jest.fn()],
    }));
    const container = shallow(<App />);
    expect(container).toMatchSnapshot();
  });

  it("app without token (with user)", () => {
    jest.resetModules();
    jest.mock("./StateProvider", () => ({
      useStateValue: () => [{ user: "adas" }, jest.fn()],
    }));
    const container = shallow(<App />);
    expect(container).toMatchSnapshot();
  });
});

describe("tests of onclick events", () => {
  it("LeftChats onclick func", () => {
    const app = shallow(<LeftChats addProp={true} />);
    // jest.mock("./LeftChats", () => ({
    //   createChat: jest.fn(() => {}),
    // }));
    const ans = app.find(".leftpart_chat").simulate("click");
    expect(ans).toMatchSnapshot();
  });

  it("simulate click on material-ui iconbutton", () => {
    const onSearchClick = sinon.spy();
    const wrapper = shallow(<Chat onSearchClick={onSearchClick} />);
    wrapper.find(IconButton).simulate("click", { preventDefault: () => {} });
    expect(onSearchClick).toMatchSnapshot();
  });
});

describe("firebase tests", () => {
  const firebase = require("firebase/app");
  firebase.firestore = firestoreMock;
  firestoreMock.reset();
  firestoreMock.collection = {
    rooms: [
      {
        roomId: "aaa",
        name: "asa",
        messages: [{ message: "asa", name: "aaa", timestamp: "2211122" }],
      },
    ],
  };
  const user = {
    displayName: "dsa",
  };

  it("function sendMessageFun test", () => {
    sendMessageFun("sdfs", "dsds", user);
    expect(firestoreMock.collection).toEqual({
      rooms: [
        {
          messages: [{ message: "asa", name: "aaa", timestamp: "2211122" }],
          name: "asa",
          roomId: "aaa",
        },
      ],
    });
  });

  it("function jwtExists test", () => {
    jwtExists("d", "sasa");
    expect(firestoreMock.collection).toEqual({
      rooms: [
        {
          messages: [{ message: "asa", name: "aaa", timestamp: "2211122" }],
          name: "asa",
          roomId: "aaa",
        },
      ],
    });
  });

  it("function send test", () => {
    send("user");
    expect(firestoreMock.collection).toEqual({
      rooms: [
        {
          messages: [{ message: "asa", name: "aaa", timestamp: "2211122" }],
          name: "asa",
          roomId: "aaa",
        },
      ],
    });
  });

  it("function idExists test", () => {
    idExists("id", "setMessages");
    expect(firestoreMock.collection).toEqual({
      rooms: [
        {
          messages: [{ message: "asa", name: "aaa", timestamp: "2211122" }],
          name: "asa",
          roomId: "aaa",
        },
      ],
    });
  });
  it("function createRoom test", () => {
    createRoom("roomName");
    expect(firestoreMock.collection).toEqual({
      rooms: [
        {
          messages: [{ message: "asa", name: "aaa", timestamp: "2211122" }],
          name: "asa",
          roomId: "aaa",
        },
      ],
    });
  });
  it("function refreshDb test", () => {
    refreshDB("setRooms");
    expect(firestoreMock.collection).toEqual({
      rooms: [
        {
          messages: [{ message: "asa", name: "aaa", timestamp: "2211122" }],
          name: "asa",
          roomId: "aaa",
        },
      ],
    });
  });
  it("service roomNameExists", () => {
    const roomId = "aaa";
    const dispatchRoomName = () => "nothing";
    const setRoomName = () => "nothing";
    const setMessages = () => "nothing";
    roomNameExists(roomId, dispatchRoomName, setRoomName, setMessages);

    db.collection("rooms")
      .doc(roomId)
      .onSnapshot((snapshot) => {
        expect(snapshot.data().name).toEqual("asa");
      });
  });
});

describe("reducer and state provider tests", () => {
  const actionTypes = {
    SET_USER: "SET_USER",
    SET_TOKEN: "SET_TOKEN",
    SET_ROOMNAME: "SET_ROOMNAME",
  };
  it("StateProvider", () => {
    const container = shallow(
      <StateProvider initialState={initialState} reducer={reducer}>
        <App />
      </StateProvider>
    );
    expect(container).toMatchSnapshot();
  });
  it("test state provider variables", () => {
    expect(StateContext).toStrictEqual(React.createContext());
    const sp = require("./StateProvider");
    sp.useStateValue();
    expect(React.useContext).toBeInstanceOf(Function);
  });
  it("reducer SET_USER", () => {
    const a = reducer("aa", { type: actionTypes.SET_USER });
    expect(a).toEqual({ 0: "a", 1: "a", user: undefined });
  });
  it("reducer without params", () => {
    const b = reducer("aa", "sss");
    expect(b).toEqual("aa");
  });
  it("reducer SET_TOKEN", () => {
    const c = reducer("aa", { type: actionTypes.SET_TOKEN });
    expect(c).toEqual({ 0: "a", 1: "a", user: undefined });
  });
  it("reducer SET_ROOMNAME", () => {
    const d = reducer("aa", { type: actionTypes.SET_ROOMNAME });
    expect(d).toEqual({ 0: "a", 1: "a", user: undefined });
  });
});
