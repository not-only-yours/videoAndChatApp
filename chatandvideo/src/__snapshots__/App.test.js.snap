// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat 1`] = `
<div
  className="chat"
>
  <div
    className="chat_header"
  >
    <WithStyles(ForwardRef(Avatar)) />
    <div
      className="chat_headerInfo"
    >
      <h3 />
      <p>
        Last seen
         
        Invalid Date
      </p>
    </div>
    <div
      className="chat_headerRight"
    >
      <WithStyles(ForwardRef(IconButton))
        onClick={[Function]}
      >
        <Memo(ForwardRef(VideoCallIcon)) />
      </WithStyles(ForwardRef(IconButton))>
    </div>
  </div>
  <div
    className="chat_body"
  />
  <div
    className="chat_footer"
  >
    <form>
      <input
        onChange={[Function]}
        placeholder="Type a message..."
        type="text"
        value=""
      />
      <button
        id="button"
        onClick={[Function]}
        type="submit"
      >
        Send a message
      </button>
    </form>
  </div>
</div>
`;

exports[`LeftChats 1`] = `
<a
  href="/rooms/undefined"
  onClick={[Function]}
>
  <div
    className="leftpart_chat"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiAvatar-fallback"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
    <div
      className="leftpartChat_info"
    >
      <h2 />
      <p />
    </div>
  </div>
</a>
`;

exports[`LeftChats addProp 1`] = `
<div
  className="leftpart_chat"
  onClick={[Function]}
>
  <h2>
    Добавити новий чатік
  </h2>
</div>
`;

exports[`LeftChats key, id, name 1`] = `
<Link
  to="/rooms/true"
>
  <div
    className="leftpart_chat"
  >
    <WithStyles(ForwardRef(Avatar)) />
    <div
      className="leftpartChat_info"
    >
      <h2>
        name
      </h2>
      <p />
    </div>
  </div>
</Link>
`;

exports[`LeftChats onclick func 1`] = `
<div
  className="leftpart_chat"
  onClick={[Function]}
>
  <h2>
    Добавити новий чатік
  </h2>
</div>
`;

exports[`Leftpart 1`] = `
<div
  className="leftpart"
>
  <div
    className="leftpart_header"
  >
    <Memo(ForwardRef(AccessibleIcon)) />
    <div
      className="leftpart_headerRight"
    >
      <WithStyles(ForwardRef(IconButton))>
        <Memo(ForwardRef(SearchIcon)) />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(IconButton))>
        <Memo(ForwardRef(InfoIcon)) />
      </WithStyles(ForwardRef(IconButton))>
    </div>
  </div>
  <div
    className="leftpart_search"
  >
    <div
      className="leftpart_searchContainer"
    >
      <Memo(ForwardRef(SearchIcon)) />
      <input
        placeholder="Print here dude"
        type="text"
      />
    </div>
  </div>
  <div
    className="leftpart_chats"
  >
    <LeftChats
      addProp={true}
    />
  </div>
</div>
`;

exports[`Login 1`] = `
<div
  className="login"
>
  <div
    className="login_container"
  >
    <img
      alt="google"
      src="https://img.icons8.com/clouds/100/000000/google-logo.png"
    />
    <div
      className="login_text"
    >
      <h1>
        Sign in to Google
      </h1>
    </div>
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
      type="submit"
    >
      Sign in With Google
    </WithStyles(ForwardRef(Button))>
  </div>
</div>
`;

exports[`StateProvider 1`] = `
<ContextProvider
  value={
    Array [
      Object {
        "roomName": null,
        "token": null,
        "user": null,
      },
      [Function],
    ]
  }
>
  <App />
</ContextProvider>
`;

exports[`Video 1`] = `
<div>
  <p
    className="roomName"
  />
  <div />
  <div />
  <div
    className="exitButton"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text exitButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Exit
      </span>
    </button>
  </div>
</div>
`;

exports[`app with user, token 1`] = `
<div
  className="app"
>
  <Video />
</div>
`;

exports[`app without token (with user) 1`] = `
<div
  className="app"
>
  <div
    className="app_body"
  >
    <BrowserRouter>
      <Leftpart />
      <Switch>
        <Route
          path="/rooms/:roomId"
        >
          <Chat />
        </Route>
        <Route
          path="/"
        >
          <Chat />
        </Route>
      </Switch>
    </BrowserRouter>
  </div>
</div>
`;

exports[`app without user 1`] = `
<div
  className="app"
>
  <Login />
</div>
`;

exports[`simulate click on material-ui iconbutton 1`] = `[Function]`;
